### PR TITLE
extracting Pipeline from SkaffoldConfig

### PR DESCRIPTION
**Pull request by balopat**
This should be a functionally equivalent change, i.e refactoring, however it does change the generated schema for the skaffold config ~also wrongly generated due to #1900 :(~

Motivation: while introducing `RunContext` in https://github.com/GoogleContainerTools/skaffold/pull/1885, it would be beneficial to just share the `Pipeline` part of the configuration, as:
 - no further functionality depends on the profiles/apiVersion/kind fields anyway 
 - the profiles struct is not serializable by gob.encoder as it embeds `yamlpatch.Node` which does not have exported fields

Update: 
#1900 is fixed now, the schema changes are correct.

### CODE REVIEW

File: pkg/skaffold/schema/latest/config.go [hunk 1]
```diff
@@ -23,19 +23,28 @@ import (
 
 const Version string = "skaffold/v1beta7"
 
-// NewSkaffoldPipeline creates a SkaffoldPipeline
+// NewSkaffoldPipeline creates a SkaffoldConfig
 func NewSkaffoldPipeline() util.VersionedConfig {
```
Comment (nkubala) [inline pkg/skaffold/schema/latest/config.go]: `NewSkaffoldConfig`?

File: pkg/skaffold/schema/latest/config.go [hunk 2]
```diff
@@ -23,19 +23,28 @@ import (
...
-	return new(SkaffoldPipeline)
+	return new(SkaffoldConfig)
 }
 
-// SkaffoldPipeline describes a Skaffold pipeline.
-type SkaffoldPipeline struct {
+// SkaffoldConfig describes a Skaffold pipeline configuration file.
```
Comment (nkubala) [inline pkg/skaffold/schema/latest/config.go]: wording is a little weird here, maybe something like 

`SkaffoldConfig holds the fields parsed from the Skaffold configuration file`?

File: docs/content/en/docs/references/yaml/main.js [hunk 3]
```diff
@@ -20,6 +20,7 @@ function* template(definitions, parentDefinition, ref, ident) {
   const allProperties = [];
   const seen = {};
 
+  console.log(name)
```
Comment (tejal29) [inline docs/content/en/docs/references/yaml/main.js]: we can remove this now.

Comment (balopat) [reply docs/content/en/docs/references/yaml/main.js]: ah yeah, thanks :)

**Comment by balopat** [comment]
I am waiting on the IDE folks to give feedback on this + we might want to merge this into `v1beta8` actually as it changes the top-level name in the config.

**Comment by balopat** [comment]
rebased + IDE folks gave a thumbs up

**Pull request merged**